backends/platform/3ds/osystem-graphics.cpp

```diff
@@ -549,14 +666,19 @@ void OSystem_3DS::displayActivityIconOnOSD(const Graphics::Surface *icon) {
 		_activityIcon.free();
 	} else {
 		if (!_activityIcon.getPixels() || icon->w != _activityIcon.w || icon->h != _activityIcon.h) {
-			_activityIcon.create(icon->w, icon->h, _pfGameTexture);
+			_activityIcon.create(icon->w, icon->h, &DEFAULT_MODE);
+		}
+
+		if (_pfGame == _gameTopTexture.format) {
```

Comment:
Is that the correct check? Shouldn't it be between the icon passed by the caller and `_activityIcon`?